commands/GitHub_Pattern_Export_To_Repo.php

```diff
@@ -0,0 +1,265 @@
+			$pattern_file_name = $pattern_file_base . '.json';
+			$json_file_path    = $category_dir . '/' . $pattern_file_name;
+
+			// Check for existing files with the same name and append a number if necessary.
+			$count = 1;
+			while ( file_exists( $json_file_path ) ) {
+				++$count;
+				$pattern_file_name = $pattern_file_base . '-' . $count . '.json';
+				$json_file_path    = $category_dir . '/' . $pattern_file_name;
+			}
+
+			// Save the pattern result to the file. Re-enconded to save as pretty JSON.
+			$result = json_decode( $result, true );
+			$result = json_encode( $result, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+			file_put_contents( $json_file_path, $result );
+
+			// Add, commit, and push the change.
+			$branch_name = 'add/pattern/' . $this->category_slug . '/' . $pattern_file_base . '-' . time();
+			\run_system_command( array( 'git', 'branch', '-m', $branch_name ), $temp_dir );
+			\run_system_command( array( 'git', 'add', $json_file_path ), $temp_dir );
+			\run_system_command( array( 'git', 'commit', '-m', 'New pattern: ' . $pattern_file_base ), $temp_dir );
+			\run_system_command( array( 'git', 'push', 'origin', $branch_name ), $temp_dir );
+
+			// Clean up by removing the cloned repository directory, if desired
+			\run_system_command( array( 'rm', '-rf', $temp_dir ), sys_get_temp_dir() );
+
+			$output->writeln( "<fg=magenta;options=bold>Pattern exported successfully to {$branch_name}.</>");
+			$output->writeln( "<fg=magenta;options=bold>View the pattern at </><fg=blue>https://github.com/a8cteam51/team51-patterns/compare/trunk...{$branch_name}</>");
+		} else {
+			$output->writeln( '<error>Pattern not found. Aborting!</error>' );
+			return Command::FAILURE;
+		}
+
+		$output->writeln( '<comment>Done!</comment>' );
+		return Command::SUCCESS;
+	}
+
+	/**
+	 * Prompts the user for a site if in interactive mode.
+	 *
+	 * @param   InputInterface  $input  The input object.
+	 * @param   OutputInterface $output The output object.
+	 *
+	 * @return  string|null
+	 */
+	private function prompt_site_input( InputInterface $input, OutputInterface $output ): ?string {
+		// Ask for the pattern name, providing an example as a hint.
+		$question = new Question( '<question>Enter the site ID or URL to extract the pattern from:</question> ' );
+		$question->setAutocompleterValues( \array_map( static fn( object $site ) => $site->url, \get_pressable_sites() ?? array() ) );
+
+		// Retrieve the user's input.
+		$site = $this->getHelper( 'question' )->ask( $input, $output, $question );
+
+		return $site ?? null;
+	}
+
+	/**
+	 * Prompts the user for a pattern name in interactive mode.
+	 *
+	 * @param InputInterface  $input  The input object.
+	 * @param OutputInterface $output The output object.
+	 *
+	 * @return string|null
+	 */
+	private function prompt_pattern_name_input( InputInterface $input, OutputInterface $output ): ?string {
+		if ( $input->isInteractive() ) {
```

Comment:
I used it on the old CLI like that but I realized that it's not actually needed when I started rebuilding the new one. So it's probably on me because people followed the pattern I laid out there 😅